test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionTextCacheTest.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class ExpressionTextCacheTest
+    {
+        private readonly ExpressionTextCache _expressionTextCache = new ExpressionTextCache();
+
+        public static IEnumerable<object[]> MemberAccessExpressions
+        {
+            get
+            {
+                return new TheoryData<Expression>
+                {
+                    (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory),
+                    (Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName),
+                    (Expression<Func<TestModel, int>>)(testmodel => testmodel.SelectedCategory.CategoryId),
+                    (Expression<Func<TestModel, string>>)(model => model.SelectedCategory.CategoryName.MainCategory)
+                };
+            }
+        }
+
+        public static IEnumerable<object[]> IndexerExpressions
+        {
+            get
+            {
+                var i = 3;
+                var key = "TestModel";
+
+                return new TheoryData<Expression>
+                {
+                    (Expression<Func<IList<TestModel>, Category>>)(model => model[2].SelectedCategory),
+                    (Expression<Func<IList<TestModel>, CategoryName>>)(testmodel => testmodel[i].SelectedCategory.CategoryName),
+                    (Expression<Func<TestModel, int>>)(model => model.PreferredCategories[i].CategoryId),
+                    (Expression<Func<IDictionary<string, TestModel>, string>>)(model => model[key].SelectedCategory.CategoryName.MainCategory),
+                    (Expression<Func<TestModel, string>>)(qmodelvar => key)
+                };
+            }
+        }
+
+        public static IEnumerable<object[]> IdenticalExpressions
+        {
+            get
+            {
+                return new TheoryData<Expression, Expression>
+                {
+                    { (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory),
+                        (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory) },
+
+                    { (Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName),
+                        (Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName) },
+
+                    { (Expression<Func<TestModel, int>>)(testmodel => testmodel.SelectedCategory.CategoryId),
+                        (Expression<Func<TestModel, int>>)(testmodel => testmodel.SelectedCategory.CategoryId) },
```

Comment:
Mix up the lambda parameter w/in these expressions e.g. `testModel => testModel` (note capital `M`) and `model => model` should be identical.